test/it/ReportingUnitAcceptanceSpec.scala

```diff
@@ -1,50 +1,132 @@
-  //        Given(s"a reporting unit exists with $TargetErn, $TargetPeriod, and $TargetRurn")
-  //        stubHBaseFor(aReportingUnitRequest(withErn = TargetErn, withPeriod = TargetPeriod, withRurn = TargetRurn).willReturn(
-  //          anOkResponse().withBody(ReportingUnitSingleMatchHBaseResponseBody)
-  //        ))
-  //
-  //        When(s"the reporting unit with $TargetErn, $TargetPeriod, and $TargetRurn is requested")
-  //        val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/reportingunits/${TargetRurn.value}").get())
-  //
-  //        Then(s"the details of the unique reporting unit identified by $TargetErn, $TargetPeriod, and $TargetRurn are returned")
-  //        response.status shouldBe OK
-  //        response.header(CONTENT_TYPE).value shouldBe JSON
-  //        response.json.as[ReportingUnit] shouldBe ReportingUnit(TargetRurn, Some("some-luref"))
-  //      }
-  //    }
+  feature("retrieve an existing Reporting Unit") {
+    scenario("by exact Enterprise reference (ERN), period, and Reporting Unit reference (RURN)") { wsClient =>
+      Given(s"a reporting unit exists with $TargetErn, $TargetPeriod, and $TargetRurn")
+      stubHBaseFor(aReportingUnitRequest(withErn = TargetErn, withPeriod = TargetPeriod, withRurn = TargetRurn).willReturn(
+        anOkResponse().withBody(ReportingUnitSingleMatchHBaseResponseBody)
+      ))
+
+      When(s"the reporting unit with $TargetErn, $TargetPeriod, and $TargetRurn is requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/reportingunits/${TargetRurn.value}").get())
+
+      Then(s"the details of the unique reporting unit identified by $TargetErn, $TargetPeriod, and $TargetRurn are returned")
+      response.status shouldBe OK
+      response.header("Content-Type") shouldBe Some(JSON)
+      response.json.as[ReportingUnit] shouldBe SampleAllValuesReportingUnit
+    }
+  }
+
+  feature("retrieve all Reporting Units that exist for an Enterprise") {
+    scenario("by exact Enterprise reference (ERN) and period") { wsClient =>
+      Given(s"multiple reporting units exist with $TargetErn, $TargetPeriod")
+      stubHBaseFor(anAllReportingUnitsForEnterpriseRequest(withErn = TargetErn, withPeriod = TargetPeriod).willReturn(
+        anOkResponse().withBody(ReportingUnitMultipleMatchHBaseResponseBody)
+      ))
+
+      When(s"the reporting units for $TargetErn, $TargetPeriod are requested")
+      val response = await(wsClient.url(s"/v1/enterprises/${TargetErn.value}/periods/${Period.asString(TargetPeriod)}/reportingunits").get())
+
+      Then(s"the details of all reporting units that are related to the Enterprise [$TargetErn] for period [$TargetPeriod] are returned")
+      response.status shouldBe OK
+      response.header("Content-Type") shouldBe Some(JSON)
+      response.json.as[Seq[ReportingUnit]] should contain theSameElementsInOrderAs
```

Comment:
Is order important here?
I would not have thought the API should be imposing ordering.